lxd/storage/backend_lxd.go

```diff
@@ -3555,11 +3560,13 @@ func (b *lxdBackend) MigrateCustomVolume(projectName string, conn io.ReadWriteCl
 	}
 
 	// Send migration index header frame with volume info and wait for receipt.
-	err = b.migrationIndexHeaderSend(l, args.IndexHeaderVersion, conn, args.Info)
+	resp, err := b.migrationIndexHeaderSend(l, args.IndexHeaderVersion, conn, args.Info)
 	if err != nil {
 		return err
 	}
 
+	args.Refresh = resp.Refresh
```

Comment:
@monstermunchkin looking at this more closely this morning, I think I see an issue here.

If the negotiated IndexHeaderVersion is 0 (i.e the target doesn't support index header) then `migrationIndexHeaderSend` will always return `resp.Refresh == false`, which will then effectively break refresh to older LXDs.

Further more even if the target server does support IndexHeaderVersion 1 then LXD 5.2 will not return a response with the `Refresh` field, meaning it will default to false, and again refresh will be broken.

One way to resolve this is to make the `Refresh` field take 3 values, by making it a pointer to a bool, then its default value `nil` means unspecified, and we can stick with the refresh mode requested.

Reply:
OK. I'll make the `Refresh` field in `InfoResponse` a pointer to a bool.

Reply:
Thanks, please could you also expand the comments here to explain what the Refresh field is and why its needed? thanks